tests/test_common_gpu.py

```diff
@@ -1124,8 +1125,8 @@ def test_serialization_shared_tensors(self):
 
     @require_non_cpu
     @pytest.mark.single_gpu_tests
+    @require_deterministic
```

Comment:
Let's add a comment here that this is required for the test to pass on XPU.

Reply:
changes to use `require_deterministic_for_xpu` for explicity